client.go

```diff
@@ -0,0 +1,179 @@
+package luasandbox
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"reflect"
+	"time"
+
+	"github.com/calyptia/api/types"
+)
+
+type Client struct {
+	url        string
+	httpClient http.Client
+	nextId     int
+}
+
+type params struct {
+	Events []types.FluentBitLogAttrs `json:"events"`
+	Filter string                    `json:"filter"`
+}
+
+type request struct {
+	JsonRpcVersion string `json:"jsonrpc"`
+	Id             int    `json:"id"`
+	Method         string `json:"method"`
+	Params         params `json:"params"`
+}
+
+type LogResult struct {
+	Log   types.FluentBitLog
+	Error string
+}
+
+type RawLogResult struct {
+	Result any    `json:"result,omitempty"`
+	Error  string `json:"error,omitempty"`
+}
+
+type response struct {
+	JsonRpcVersion string         `json:"jsonrpc"`
+	Id             int            `json:"id"`
+	Result         []RawLogResult `json:"result,omitempty"`
+	Error          *rpcError      `json:"error,omitempty"`
+}
+
+type rpcError struct {
+	Code    int    `json:"code"`
+	Message string `json:"message"`
+}
+
+func New(url string) *Client {
+	rv := &Client{
+		url:    url,
+		nextId: 1,
+		httpClient: http.Client{
+			Transport: &http.Transport{
+				MaxIdleConns:       10,
+				IdleConnTimeout:    30 * time.Second,
+				DisableCompression: true,
+			},
+		},
+	}
+
+	return rv
+}
+
+func (c *Client) Run(events []types.FluentBitLog, filter string) ([]LogResult, error) {
```

Comment:
```suggestion
func (c *Client) Run(ctx context.Context, events []types.FluentBitLog, filter string) ([]LogResult, error) {
```

Please add `context.Context` and build the request using it.

```go
http.NewRequestWithContext(ctx, http.MethodPost. c.url, bytes.NewReader(reqBody))
```